tests/modules/basics.c

```diff
@@ -407,6 +407,26 @@ int TestStringAppendAM(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
     return REDISMODULE_OK;
 }
 
+/* TEST.STRING.TRIM -- Test we trim a string with free space. */
+int TestTrimString(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    REDISMODULE_NOT_USED(argv);
+    REDISMODULE_NOT_USED(argc);
+    RedisModuleString *s = RedisModule_CreateString(ctx,"foo",3);
+    char *tmp = RedisModule_Alloc(1024);
+    RedisModule_StringAppendBuffer(ctx,s,tmp,1024);
+    size_t string_len = RedisModule_MallocSizeString(s);
+    RedisModule_TrimStringAllocation(s);
+    size_t len_after_trim = RedisModule_MallocSizeString(s);
+    if (len_after_trim < string_len) {
+        RedisModule_ReplyWithSimpleString(ctx, "OK");
+    } else {
+        RedisModule_Log(ctx, "Error", "String was not trimmed as expected.");
```

Comment:
logging isn't enough, no one will see it, to fail the test this function should return 0 and 1 so that the assertion of where it's called can get triggered.

Reply:
I followed the file's tests logic, It will fail in `if (!TestAssertStringReply(ctx,reply,"OK",2)) goto fail;` . I validate it fails with the following error: 
```
[err]: test module api basics in tests/unit/moduleapi/basics.tcl
Expected 'ALL TESTS PASSED' to equal or match 'SOME TEST DID NOT PASS! Check server logs'
```

Reply:
maybe i didn't read this test properly the previous time.
this function is a registered command right, it has one flow that replies OK, and another flow that doesn't reply (only logs an error), in theory, if this command is called from TCL client, rather than RM_Call, it can cause the client to hang.
i think it should `RM_ReplyWithError` in the second flow.

Reply:
changed to `RM_ReplyWithError`